service/history/replication/task_ack_manager.go

```diff
@@ -101,16 +106,21 @@ func NewTaskAckManager(
 	retryPolicy.SetMaximumAttempts(config.ReplicatorReadTaskMaxRetryCount())
 	retryPolicy.SetBackoffCoefficient(1)
 
+	taskCreatedTime := atomic.Value{}
+	taskCreatedTime.Store(time.Now())
```

Comment:
nit: use `shard.GetTimesource().Now()`?

If there's a replication backlog on startup, the batch size for the first response will likely to be small I think? Probably we can always use defaultBatchSize for the first response by storing an empty time here?

Reply:
make sense. use default batch size on first response